### DISCUSSION TITLE
How to include all language URLs in Sitemap?

### DISCUSSION THREAD

**Discussion by enzedonline** [wagtail/wagtail-localize#392] [Q&A]
Using the contrib.sitemaps, it seems to only generate sitemap for the default language.

Is there a way to get it to index the other locales as well?

**Comment by enzedonline** [comment]
I added a custom get_sitemap_urls to the home page model - does the job but maybe there is a better way to include other locales? This way makes a few presumptions about your site layout:

    def get_sitemap_urls(self, request):
        sitemap = super().get_sitemap_urls(request)

        for locale_home in self.get_siblings(inclusive=False).live():
            for entry in locale_home.get_sitemap_urls(request):
                sitemap.append(entry)
            for child_page in locale_home.get_descendants().live():
                for entry in child_page.get_sitemap_urls(request):
                    sitemap.append(entry)
        return sitemap

**Reply by kaedroho** [reply]
Thanks for reporting! This is currently an unimplemented feature, see https://github.com/wagtail/wagtail-localize/issues/184

The plan is to add x-link tags into the existing sitemap, so Google recognises those pages as translations

**Comment by ViggieM** [comment]
I use a custom Sitemap class for generating sitemaps in multilingual sites
```python
from wagtail.contrib.sitemaps import Sitemap


class CustomSitemap(Sitemap):
    def items(self):
        root_page = self.get_wagtail_site().root_page
        items = (
            root_page.get_descendants(inclusive=True).live().public().order_by("path").defer_streamfields().specific()
        )
        root_translations = root_page.get_translations()
        for translation in root_translations:
            # gets also any other translated pages that exist in other languages, but not under the root page
            translation_items = (
                translation.get_descendants(inclusive=False)
                .live()
                .public()
                .order_by("path")
                .defer_streamfields()
                .specific()
                .exclude(translation_key__in=items.values_list("translation_key", flat=True))
            )
            items |= translation_items
        return items
```
and in `urls.py`
```
sitemaps = {"wagtail": CustomSitemap}
urlpatterns= [
    path("sitemap.xml", sitemap, {"sitemaps": sitemaps}, name="sitemap"),
]
```
